### PR TITLE
Propagate query params in Databricks notebook/job links

### PR DESCRIPTION
**Pull request by smurching**
Fixes query parameter string propagation in links to Databricks notebooks / jobs.

Also upgrades create-react-app dependency to 2.0 (as recommended [here](https://facebook.github.io/create-react-app/docs/troubleshooting#npm-run-build-fails-to-minify)) to support libraries using modern JS syntax (like the popular [query-string](https://www.npmjs.com/package/query-string) library) & modern APIs (like [URL](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL)), updating package.json accordingly.

### CODE REVIEW

File: mlflow/server/js/package.json [hunk 1]
```diff
@@ -81,13 +83,10 @@
       "lcov"
     ]
   },
-  "proxy": {
```
Comment (smurching) [inline mlflow/server/js/package.json]: Moved to `src/setupProxy.js` as part of migration to create-react-app 2.0 ([see create-react-app docs](https://facebook.github.io/create-react-app/docs/proxying-api-requests-in-development#configuring-the-proxy-manually))

File: mlflow/server/js/package.json [hunk 2]
```diff
@@ -81,13 +83,10 @@
...
-    "/ajax-api": {
-      "target": "http://localhost:5000"
-    },
-    "/get-artifact": {
-      "target": "http://localhost:5000",
-      "ws": true
-    }
-  }
+  "browserslist": [
```
Comment (smurching) [inline mlflow/server/js/package.json]: Added by browserslist, which is a transitive dependency of react-scripts 2.0 ([see docs](https://facebook.github.io/create-react-app/docs/post-processing-css)).

File: mlflow/server/js/package.json [hunk 3]
```diff
@@ -59,13 +60,14 @@
...
+    "jest-localstorage-mock": "^2.4.0",
+    "react-app-rewire-define-plugin": "1.0.0",
+    "react-app-rewired": "^2.1.1",
+    "webpack": "^4.28.3"
   },
   "scripts": {
     "start": "react-app-rewired start",
-    "build": "react-app-rewired build",
+    "build": "react-app-rewired --max_old_space_size=4096 build",
     "test": "react-app-rewired test --env=jsdom",
```
Comment (andrewmchen) [inline mlflow/server/js/package.json]: Should we do this? https://github.com/facebook/create-react-app/blob/master/CHANGELOG.md#the-default-jest-environment-was-changed-to-jsdom

File: mlflow/server/js/src/utils/Utils.test.js [hunk 4]
```diff
@@ -153,7 +153,19 @@ test("formatSource & renderSource", () => {
   };
   const wrapper2 = shallow(Utils.renderSource(databricksRun, databricksRunRevisionTags));
   expect(wrapper2.is("a")).toEqual(true);
-  expect(wrapper2.props().href).toEqual("../#notebook/13/revision/42");
+  expect(wrapper2.props().href).toEqual("http://localhost/#notebook/13/revision/42");
```
Comment (andrewmchen) [inline mlflow/server/js/src/utils/Utils.test.js]: micronit: maybe it's preferable to use these matchers? https://github.com/producthunt/chai-enzyme

File: mlflow/server/js/src/utils/Utils.test.js [hunk 5]
```diff
@@ -153,7 +153,19 @@ test("formatSource & renderSource", () => {
...
+
+  const wrapper3 = shallow(Utils.renderSource(databricksRun, databricksRunRevisionTags, "?o=123"));
+  expect(wrapper3.is("a")).toEqual(true);
+  // Query params must appear before the hash, see https://tools.ietf.org/html/rfc3986#section-4.2
+  // and https://stackoverflow.com/a/34772568
+  expect(wrapper3.props().href).toEqual("http://localhost/?o=123#notebook/13/revision/42");
+});
+
+test("addQueryParams", () => {
+  expect(Utils.addQueryParams("http://localhost/foo", "?o=123")).toEqual("http://localhost/foo?o=123");
```
Comment (andrewmchen) [inline mlflow/server/js/src/utils/Utils.test.js]: I think in the future, we should aim to use these.. 
https://github.com/producthunt/chai-enzyme

**Comment by smurching** [comment]
Thanks @andrewmchen, will merge to master - also thanks for the tip RE the chai-enzyme matchers, I guess the main advantage is that you can use chai-style assertions in your enzyme tests (based on [this page](https://airbnb.io/enzyme/#running-enzyme-tests))?

**Pull request merged**